### PR TITLE
refactor: Remove Enterprise API V1

### PR DESCRIPTION
**Pull request by ruzniaievdm**
Replace these [usage1](https://github.com/openedx/ecommerce/blob/master/ecommerce/enterprise/utils.py#L152) [usage2](https://github.com/openedx/ecommerce/blob/master/ecommerce/enterprise/utils.py#L515-L560) with corresponding to the enterprise-catalog service

YT: https://youtrack.raccoongang.com/issue/OeX_Depr-249

### CODE REVIEW

**Review by idegtiarov — APPROVED** [review]
LGTM

**Comment by idegtiarov** [comment]
But please take a look at tests

**Review by dyudyunov — APPROVED** [review]

File: ecommerce/enterprise/tests/mixins.py [hunk 1]
```diff
@@ -79,11 +76,11 @@ def mock_enterprise_customer_list_api_get(self):
 
     def mock_enterprise_catalog_api_get(self, enterprise_catalog_uuid, custom_response=None):
         """
-        Helper function to register the legacy enterprise catalog API endpoint using responses.
+        Helper function to register the enterprise catalog API endpoint using responses.
         """
         enterprise_catalog_api_response = {
             "count": 60,
-            "next": "{}{}/?page=2".format(self.LEGACY_ENTERPRISE_CATALOG_URL, enterprise_catalog_uuid),
+            "next": f'{self.ENTERPRISE_CATALOG_URL}{enterprise_catalog_uuid}/get_content_metadata/?page=2',
```
Comment (idegtiarov) [inline ecommerce/enterprise/tests/mixins.py]: urljoin would be better or consistency

File: ecommerce/enterprise/tests/mixins.py [hunk 2]
```diff
@@ -564,24 +561,29 @@ def mock_enterprise_catalog_api(self, enterprise_customer_uuid, raise_exception=
         Helper function to register the enterprise catalog API endpoint.
         """
         enterprise_catalog_api_response = {
-            'count': 10,
-            'num_pages': 3,
-            'current_page': 2,
+            'count': 2,
+            'next': f'{self.ENTERPRISE_CATALOG_URL}?enterprise_customer={enterprise_customer_uuid}',
```
Comment (idegtiarov) [inline ecommerce/enterprise/tests/mixins.py]: ditto

File: ecommerce/enterprise/utils.py [hunk 3]
```diff
@@ -529,42 +529,29 @@ def has_enterprise_offer(basket):
...
-        limit (int): The number of results to return per page.
+        page_size (int): The number of results to return per page.
         page (int): The page number to fetch.
         endpoint_request_url (str): This is used to replace the lms url with ecommerce url
 
     Returns:
         dict: The result set containing the content objects associated with the Enterprise Catalog.
         NoneType: Return None if no catalog with that uuid is found.
-
-    Response will look like
```
Comment (idegtiarov) [inline ecommerce/enterprise/utils.py]: Could we leave an example of the response in the docstrings of course with the actual data?

File: ecommerce/enterprise/utils.py [hunk 4]
```diff
@@ -529,42 +529,29 @@ def has_enterprise_offer(basket):
...
-            "content_last_modified": None,
-            "catalog_modified": "2022-05-26T12:56:20.346837Z",
-            "query_title": None
-        }
-
     """
     resource = 'enterprise-catalogs'
     partner_code = site.siteconfiguration.partner.short_code
-    cache_key = u'{site_domain}_{partner_code}_{resource}_{catalog}_{limit}_{page}'.format(
+    cache_key = u'{site_domain}_{partner_code}_{resource}_{catalog}_{page_size}_{page}'.format(
```
Comment (idegtiarov) [inline ecommerce/enterprise/utils.py]: u  - prefix could be omitted since we are on py3.x or even better we could use f-string instead of formated one.

**Review by idegtiarov — CHANGES_REQUESTED** [review]
Thank you for the PR, please take a look at some comments, nits, and questions.

File: ecommerce/enterprise/utils.py (hunk 3 above)
Comment (ruzniaievdm) [reply ecommerce/enterprise/utils.py]: I think about it but the response with actual data takes a lot of lines

Comment (idegtiarov) [reply ecommerce/enterprise/utils.py]: That should not stop you from doing it

Comment (ruzniaievdm) [reply ecommerce/enterprise/utils.py]: ok, in that case i will add

File: ecommerce/enterprise/tests/mixins.py (hunk 1 above)
Comment (ruzniaievdm) [reply ecommerce/enterprise/tests/mixins.py]: done

**Comment by idegtiarov** [comment]
@ruzniaievdm Please take a look at the commit message before you will open PR to upstream, now Lint Commit Messages checker is not passed

**Review by idegtiarov — APPROVED** [review]

**Comment by ruzniaievdm** [comment]
@idegtiarov  Thanks for review, and what about commit message, this will be fixed text after rebasing to one commit

**Comment by idegtiarov** [comment]
👍🏻

**Review by idegtiarov — APPROVED** [review]
Thanks, LGTM!

**Pull request closed without merging**